adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,48 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    std::stack<char> bracketsStack;
+    std::string close;
+    std::string open;
+
+    for(auto bracket : brackets)
+    {
+        open+=bracket.first;
+        close+=bracket.second;
+    }
+
+    for(auto character : source)
+    {
+        if(open.find(character) != std::string::npos)
+        {
+            bracketsStack.push(character);
+        }
+        else if(close.find(character) != std::string::npos)
+        {
+            if(bracketsStack.empty())
+                return false;
+            for(auto bracket : brackets)
+            {
+                if(bracket.second == character)
+                {
+                    if(bracketsStack.top() != bracket.first)
```

Comment:
please be more explicit with your conditions.
use const bool for that - if (const bool isOpened = ...)/if (const bool bracketsMatch = ..) 
that may improve code readability